sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1201,62 +1144,68 @@ auto node = graph.add([&](sycl::handler& cgh){
+will throw synchronously with error code `invalid`.
 
-Design 1) has the drawback of forcing users to keep the user pointer variable
-alive so that the reference is valid, which is unintuitive and is likely to
-result in bugs.
+- Waiting on a queue in the recording state is an error and will throw
+synchronously with error code `invalid`.
 
-Design 2) introduces a handle object which has the advantages of being a less
-error prone way to provide the pointer to the deferred allocation. However, it
-requires kernel changes and introduces an overhead above the raw pointers that
-are the advantage of USM.
+==== Buffer Limitations
 
-Design 3) needs specific backend support for deferred allocation.
+Because of the delayed execution of a recorded graph, it is not possible to support
+captured code which relies on the copy-back on destruction behavior of buffers.
+Typically, applications would rely on this behavior to do work on the host which
+cannot inherently be captured inside a command graph.
 
-=== Device Specific Graph
+- Thus, when recording to a graph it is an error to submit a command which has
+an accessor on a buffer which would cause a write-back to happen. Using an
+incompatible buffer in this case will result in a synchronous error being
+thrown with error code `invalid`.
 
-A modifiable state `command_graph` contains nodes targeting specific devices,
-rather than being a device agnostic representation only tied to devices on
-finalization. This allows the implementation to process nodes which require
-device information when the command group function is evaluated. For example,
-a SYCL reduction implementation may desire the work-group/sub-group size, which
-is normally gathered by the runtime from the device associated with the queue.
+- The copy-back mechanism can be disabled explicitly for buffers with attached host
+storage using either `buffer::set_final_data(nullptr)` or
+`buffer::set_write_back(false)`.
 
-This design also enables the future capability for a user to compose a graph
-with nodes targeting different devices, allowing the benefits of defining an
-execution graph ahead of submission to be extended to multi-device platforms.
-Without this capability a user currently has to submit individual single-device
-graphs and use events for dependencies, which is a usage model this extension is
-aiming to optimize. Automatic load balancing of commands across devices is not a
-problem this extension currently aims to solve, it is the responsibility of the
-user to decide the device each command will be processed for, not the SYCL
-runtime.
+- It is also an error to create a host accessor to a buffer which is used in
+commands which are currently being recorded to a command graph. Attempting to
+construct a host accessor to an incompatible buffer will result in a
+synchronous error being thrown with error code `invalid`.
+
+==== Error Handling
+
+- When a queue is in recording mode asynchronous exceptions will not be
```

Comment:
This looks slightly odd as a single bullet-point that's a full paragraph, would remove the bullet-point